pkg/template/controller/templateinstance_controller.go

```diff
@@ -307,10 +311,35 @@ func (c *TemplateInstanceController) instantiate(templateInstance *templateapi.T
 			RESTMapper:  client.DefaultMultiRESTMapper(),
 			ObjectTyper: kapi.Scheme,
 			ClientMapper: resource.ClientMapperFunc(func(mapping *meta.RESTMapping) (resource.RESTClient, error) {
+				config := *c.config
+
+				// TODO copied from pkg/cmd/util/clientcmd/factory_object_mapping.go#ClientForMapping
```

Comment:
Why can't you move this to an openshift `/resource` package that returns a clientmapperfunc based on a config?